ray-operator/controllers/raycluster_controller.go

```diff
@@ -165,11 +165,13 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayiov1alpha1.RayCluster)
 		}
 	} else if len(headPods.Items) > 1 {
 		log.Info("reconcilePods ", "more than 1 head pod found for cluster", instance.Name)
-		for index := range headPods.Items {
```

Comment:
I think that's the problem. We should not change `headPods.Items` during the iteration. I wrote the code snippet to verify the change. It looks good to me.

https://go.dev/play/p/sI7CyX6KmrK